specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/stable/2020-03-01/outputs.json

```diff
@@ -0,0 +1,1074 @@
+        },
+        "propertyColumns": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "AzureSqlDatabaseOutputDataSource": {
+      "description": "Describes an Azure SQL database output data source.",
+      "x-ms-discriminator-value": "Microsoft.Sql/Server/Database",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OutputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AzureSqlDatabaseOutputDataSourceProperties",
+          "description": "The properties that are associated with an Azure SQL database output. Required on PUT (CreateOrReplace) requests.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "AzureSqlDatabaseOutputDataSourceProperties": {
+      "description": "The properties that are associated with an Azure SQL database output.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureSqlDatabaseDataSourceProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "AzureSynapseOutputDataSource": {
+      "description": "Describes an Azure Synapse output data source.",
+      "x-ms-discriminator-value": "Microsoft.Sql/Server/DataWarehouse",
+      "allOf": [
+        {
+          "$ref": "#/definitions/OutputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AzureSynapseOutputDataSourceProperties",
+          "description": "The properties that are associated with an Azure Synapse output. Required on PUT (CreateOrReplace) requests.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "AzureSynapseOutputDataSourceProperties": {
+      "description": "The properties that are associated with an Azure Synapse output.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AzureSynapseDataSourceProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "AzureSynapseDataSourceProperties": {
+      "description": "The properties that are associated with an Azure SQL database data source.",
+      "properties": {
+        "server": {
+          "type": "string",
+          "description": "The name of the SQL server containing the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "database": {
+          "type": "string",
+          "description": "The name of the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "table": {
+          "type": "string",
+          "description": "The name of the table in the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "user": {
+          "type": "string",
+          "description": "The user name that will be used to connect to the Azure SQL database. Required on PUT (CreateOrReplace) requests."
+        },
+        "password": {
```

Comment:
>password [](http://example.com/codeflow?start=9&length=8)

Tag this with x-ms-secret extension

Reply:
Hi, can you link me to the documentation for this extension? I don't see it here, https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md, is this doc out of date?

Reply:
Yes, x-ms-secret should be documented there. Anyway, it's just a boolean flag used to mark secrets:
"x-ms-secret": true. I created this issue to get the documentation updated: [https://github.com/Azure/autorest/issues/4115](https://github.com/Azure/autorest/issues/4115)